catalogs/catalog-fileset/src/main/java/org/apache/gravitino/catalog/fileset/FilesetCatalogOperations.java

```diff
@@ -137,6 +140,23 @@ public class FilesetCatalogOperations extends ManagedSchemaOperations
   @VisibleForTesting ScheduledThreadPoolExecutor scheduler;
   @VisibleForTesting Cache<FileSystemCacheKey, FileSystem> fileSystemCache;
 
+  private final ThreadPoolExecutor fileSystemExecutor =
+      new ThreadPoolExecutor(
+          Math.max(2, Math.min(Runtime.getRuntime().availableProcessors() * 2, 16)),
```

Comment:
Do we need to keep such threads active for use? My feeling is that most of the threads can be swept when idle to save resources.

Reply:
I have allowed core poll timeout, and the threads will stop if there is no task assigned to the pool.